spaceinvaders.py

```diff
@@ -1,648 +1,687 @@
+        self.makeNewShip = False
+        self.shipAlive = True
+
+    def make_blockers(self, number):
+        blockerGroup = sprite.Group()
+        for row in range(4):
+            for column in range(9):
+                blocker = Blocker(10, GREEN, row, column)
+                blocker.rect.x = 50 + (200 * number) + (column * blocker.width)
+                blocker.rect.y = 450 + (row * blocker.height)
+                blockerGroup.add(blocker)
+        return blockerGroup
+
+    def reset_lives_sprites(self):
+        self.life1 = Life(715, 3)
+        self.life2 = Life(742, 3)
+        self.life3 = Life(769, 3)
+
+        if self.lives == 3:
+            self.livesGroup = sprite.Group(self.life1, self.life2, self.life3)
+        elif self.lives == 2:
+            self.livesGroup = sprite.Group(self.life1, self.life2)
+        elif self.lives == 1:
+            self.livesGroup = sprite.Group(self.life1)
+
+    def reset_lives(self, lives):
+        self.lives = lives
+        self.reset_lives_sprites()
+
+    def create_audio(self):
+        self.sounds = {}
+        for sound_name in ["shoot", "shoot2", "invaderkilled", "mysterykilled",
+                           "shipexplosion"]:
+            self.sounds[sound_name] = mixer.Sound(
+                SOUND_PATH + "{}.wav".format(sound_name))
+            self.sounds[sound_name].set_volume(0.2)
+
+        self.musicNotes = [mixer.Sound(SOUND_PATH + "{}.wav".format(i)) for i
+                           in range(4)]
+        for sound in self.musicNotes:
+            sound.set_volume(0.5)
+
+        self.noteIndex = 0
+
+    def play_main_music(self, currentTime):
+        moveTime = self.enemies.sprites()[0].moveTime
+        if currentTime - self.noteTimer > moveTime:
+            self.note = self.musicNotes[self.noteIndex]
+            if self.noteIndex < 3:
+                self.noteIndex += 1
+            else:
+                self.noteIndex = 0
+
+            self.note.play()
+            self.noteTimer += moveTime
+
+    def create_text(self):
+        self.titleText = Text(FONT, 50, "Space Invaders", WHITE, 164, 155)
+        self.titleText2 = Text(FONT, 25, "Press any key to continue", WHITE,
+                               201, 225)
+        self.gameOverText = Text(FONT, 50, "Game Over", WHITE, 250, 270)
+        self.nextRoundText = Text(FONT, 50, "Next Round", WHITE, 240, 270)
+        self.enemy1Text = Text(FONT, 25, "   =   10 pts", GREEN, 368, 270)
+        self.enemy2Text = Text(FONT, 25, "   =  20 pts", BLUE, 368, 320)
+        self.enemy3Text = Text(FONT, 25, "   =  30 pts", PURPLE, 368, 370)
+        self.enemy4Text = Text(FONT, 25, "   =  ?????", RED, 368, 420)
+        self.scoreText = Text(FONT, 20, "Score", WHITE, 5, 5)
+        self.livesText = Text(FONT, 20, "Lives ", WHITE, 640, 5)
+
+    def check_input(self):
+        self.keys = key.get_pressed()
+        for e in event.get():
+            if e.type == QUIT or (e.type == KEYUP and e.key == K_ESCAPE):
```

Comment:
Since this was added in 3 places, this could be moved to a function called something like `should_exit()`.

Reply:
'Refactor exit condition w should_exit function' (093850f)